gcp-observability/src/main/java/io/grpc/gcp/observability/ObservabilityConfigImpl.java

```diff
@@ -74,95 +87,128 @@ void parse(String config) throws IOException {
-        enableCloudLogging = value;
-      }
-      value = JsonUtil.getBoolean(config, "enable_cloud_monitoring");
-      if (value != null) {
-        enableCloudMonitoring = value;
-      }
-      value = JsonUtil.getBoolean(config, "enable_cloud_trace");
-      if (value != null) {
-        enableCloudTracing = value;
-      }
-      destinationProjectId = JsonUtil.getString(config, "destination_project_id");
-      flushMessageCount = JsonUtil.getNumberAsLong(config, "flush_message_count");
-      List<?> rawList = JsonUtil.getList(config, "log_filters");
-      if (rawList != null) {
-        List<Map<String, ?>> jsonLogFilters = JsonUtil.checkObjectList(rawList);
-        ImmutableList.Builder<LogFilter> logFiltersBuilder = new ImmutableList.Builder<>();
-        for (Map<String, ?> jsonLogFilter : jsonLogFilters) {
-          logFiltersBuilder.add(parseJsonLogFilter(jsonLogFilter));
-        }
-        this.logFilters = logFiltersBuilder.build();
+      projectId = JsonUtil.getString(config, "project_id");
+      parseLoggingObject(config, CONFIG_LOGGING_VAR_NAME);
+      parseMonitoringObject(config, CONFIG_MONITORING_VAR_NAME);
+      parseTracingObject(config, CONFIG_TRACING_VAR_NAME);
+      parseCustomTags(config, CONFIG_LABELS_VAR_NAME);
+    }
+  }
+
+  private void parseLoggingObject(Map<String, ?> config,  String jsonObjectName) {
+    Map<String, ?> rawCloudLogging = JsonUtil.getObject(config, jsonObjectName);
+    if (rawCloudLogging != null) {
+      enableCloudLogging = true;
+      List<?> rawClientRpcEvents = JsonUtil.getList(rawCloudLogging, "client_rpc_events");
+      clientLogFilters =
+          rawClientRpcEvents != null ? parseRpcEvents(rawClientRpcEvents) : clientLogFilters;
+      List<?> rawServerRpcEvents = JsonUtil.getList(rawCloudLogging, "server_rpc_events");
+      serverLogFilters =
+          rawServerRpcEvents != null ? parseRpcEvents(rawServerRpcEvents) : serverLogFilters;
+    }
+  }
+
+  private void parseMonitoringObject(Map<String, ?> config, String jsonObjectName) {
+    Map<String, ?> rawCloudMonitoring = JsonUtil.getObject(config, jsonObjectName);
+    if (rawCloudMonitoring != null) {
+      if (rawCloudMonitoring.isEmpty()) {
+        enableCloudMonitoring = true;
       }
-      rawList = JsonUtil.getList(config, "event_types");
-      if (rawList != null) {
-        List<String> jsonEventTypes = JsonUtil.checkStringList(rawList);
-        ImmutableList.Builder<EventType> eventTypesBuilder = new ImmutableList.Builder<>();
-        for (String jsonEventType : jsonEventTypes) {
-          eventTypesBuilder.add(convertEventType(jsonEventType));
+    }
+  }
+
+  private void parseTracingObject(Map<String, ?> config, String jsonObjectName) {
+    Map<String, ?> rawCloudTracing = JsonUtil.getObject(config, jsonObjectName);
+    if (rawCloudTracing != null) {
+      enableCloudTracing = true;
+      this.sampler = Samplers.probabilitySampler(0.0);
+      if (!rawCloudTracing.isEmpty()) {
```

Comment:
What is with this check?

Reply:
Added this check the intention of adding non-default sampler when specified. Now that you pointed, I see it is redundant. Will remove this check.